pkg/fleets/controller_test.go

```diff
@@ -360,49 +519,72 @@ func TestControllerRun(t *testing.T) {
+		gsSet2.Status.AllocatedReplicas = 2
+
+		m.AgonesClient.AddReactor("list", "gameserversets",
+			func(action k8stesting.Action) (bool, runtime.Object, error) {
+				return true, &agonesv1.GameServerSetList{Items: []agonesv1.GameServerSet{*gsSet1, *gsSet2}}, nil
+			})
 
-	gsSet2 := fleet.GameServerSet()
-	// nolint:goconst
-	gsSet2.ObjectMeta.Name = "gsSet2"
-	gsSet2.Status.Replicas = 5
-	gsSet2.Status.ReadyReplicas = 5
-	gsSet2.Status.ReservedReplicas = 3
-	gsSet2.Status.AllocatedReplicas = 2
+		updated := false
+		m.AgonesClient.AddReactor("update", "fleets",
+			func(action k8stesting.Action) (bool, runtime.Object, error) {
+				updated = true
+				ua := action.(k8stesting.UpdateAction)
+				fleet := ua.GetObject().(*agonesv1.Fleet)
 
-	m.AgonesClient.AddReactor("list", "gameserversets",
-		func(action k8stesting.Action) (bool, runtime.Object, error) {
-			return true, &agonesv1.GameServerSetList{Items: []agonesv1.GameServerSet{*gsSet1, *gsSet2}}, nil
-		})
+				assert.Equal(t, gsSet1.Status.Replicas+gsSet2.Status.Replicas, fleet.Status.Replicas)
+				assert.Equal(t, gsSet1.Status.ReadyReplicas+gsSet2.Status.ReadyReplicas, fleet.Status.ReadyReplicas)
+				assert.Equal(t, gsSet1.Status.ReservedReplicas+gsSet2.Status.ReservedReplicas, fleet.Status.ReservedReplicas)
+				assert.Equal(t, gsSet1.Status.AllocatedReplicas+gsSet2.Status.AllocatedReplicas, fleet.Status.AllocatedReplicas)
+				return true, fleet, nil
+			})
 
-	updated := false
-	m.AgonesClient.AddReactor("update", "fleets",
-		func(action k8stesting.Action) (bool, runtime.Object, error) {
-			updated = true
-			ua := action.(k8stesting.UpdateAction)
-			fleet := ua.GetObject().(*agonesv1.Fleet)
+		_, cancel := agtesting.StartInformers(m, c.fleetSynced, c.gameServerSetSynced)
+		defer cancel()
 
-			assert.Equal(t, gsSet1.Status.Replicas+gsSet2.Status.Replicas, fleet.Status.Replicas)
-			assert.Equal(t, gsSet1.Status.ReadyReplicas+gsSet2.Status.ReadyReplicas, fleet.Status.ReadyReplicas)
-			assert.Equal(t, gsSet1.Status.ReservedReplicas+gsSet2.Status.ReservedReplicas, fleet.Status.ReservedReplicas)
-			assert.Equal(t, gsSet1.Status.AllocatedReplicas+gsSet2.Status.AllocatedReplicas, fleet.Status.AllocatedReplicas)
-			return true, fleet, nil
-		})
+		err := c.updateFleetStatus(fleet)
+		assert.Nil(t, err)
+		assert.True(t, updated)
+	})
 
-	_, cancel := agtesting.StartInformers(m, c.fleetSynced, c.gameServerSetSynced)
-	defer cancel()
+	t.Run("list gameservers returns an error", func(t *testing.T) {
+		fleet := defaultFixture()
+		c, _ := newFakeController()
+		c.gameServerSetLister = &fakeGSListerWithErr{}
+
+		err := c.updateFleetStatus(fleet)
+		assert.EqualError(t, err, "error listing gameserversets for fleet fleet-1: random-err")
+	})
+
+	t.Run("fleets getter returns an error", func(t *testing.T) {
+		fleet := defaultFixture()
+		c, _ := newFakeController()
+
+		c.fleetGetter = &fakeFleetsGetterWithErr{}
+
+		err := c.updateFleetStatus(fleet)
+
```

Comment:
I think there is no need to have blank line here. And in some other places too.

Reply:
I'm following unit testing arrange - act - assert pattern and I prefer to separate each step with a blank line at least.